### PR TITLE
feat: Implement --test-names/--name for dev/test commands

### PR DESCRIPTION
**Pull request by 10ko**

### CODE REVIEW

**Comment by eysi09** [comment]
@10ko could you outline the design you had in mind? Since this is core framework stuff we should make sure we're all on the same page (cc @edvald).

**Comment by 10ko** [comment]
Hi @eysi09, I still don’t have something 100% clear in mind, this is a very early stage WIP PR. I will post some more comments and questions in #851 and then outline the design here. ✌️

File: docs/reference/commands.md [hunk 1]
```diff
@@ -189,9 +189,11 @@ as you modify the code.
 Examples:
 
     garden dev
-    garden dev --hot-reload=foo-service       # enable hot reloading for foo-service
     garden dev --hot=foo-service,bar-service  # enable hot reloading for foo-service and bar-service
     garden dev --hot=*                        # enable hot reloading for all compatible services
+    garden dev --skip-tests=foo-service       # skip running any tests
```
Comment (edvald) [inline docs/reference/commands.md]: The parameter shouldn't have a value

File: garden-service/src/commands/dev.ts [hunk 2]
```diff
@@ -51,6 +51,11 @@ const devOpts = {
   "skip-tests": new BooleanParameter({
     help: "Disable running the tests",
   }),
+  "test-names": new StringsParameter({
+    help: "The name(s) of the module(s) to test (skip to test all modules). " +
```
Comment (edvald) [inline garden-service/src/commands/dev.ts]: The help string is misleading. Shouldn't it be more like "Filter the tests to run by test name (leave unset to run all tests). "? Phrasing suggestion @thsig?

File: garden-service/test/unit/src/commands/test.ts [hunk 3]
```diff
@@ -68,4 +68,45 @@ describe("commands.test", () => {
       },
     })).to.be.true
   })
+
+  it.only("should only run integration tests if the option 'name' is specified with a glob", async () => {
```
Comment (edvald) [inline garden-service/test/unit/src/commands/test.ts]: Oops, left an `.only` here!

File: garden-service/src/commands/dev.ts [hunk 4]
```diff
@@ -51,6 +51,11 @@ const devOpts = {
...
+      "Accepts glob patterns (e.g. integ* would run both 'integ' and 'integration')",
```
Comment (edvald) [inline garden-service/src/commands/dev.ts]: Also, missing period at end. Maybe add one for the skip-tests parameter above as well.

File: garden-service/test/unit/src/commands/test.ts (hunk 3 above)
Comment (10ko) [reply garden-service/test/unit/src/commands/test.ts]: 🤦‍♂

File: docs/reference/module-types/exec.md [hunk 5]
```diff
@@ -353,58 +351,3 @@ tests:
     command:
     env: {}
 ```
-
-## Outputs
```
Comment (edvald) [inline docs/reference/module-types/exec.md]: Hmm, maybe you need to rebase or re-run docs generation. This shouldn't disappear.

File: garden-service/src/commands/dev.ts (hunk 2 above)
Comment (thsig) [reply garden-service/src/commands/dev.ts]: How about this?

"Filter the tests to run by test name across all modules (leave unset to run all tests)."

I.e. to make explicit that this filters across modules (which implicitly recommends using common names like `unit` or `integ`).

**Pull request merged**